### PR TITLE
GH-40775: [Benchmarking][Java] Fix conbench timeout

### PR DESCRIPTION
**Pull request by danepitkin**
### Rationale for this change

The java build script has been recently updated and it is affecting conbench, which is now seeing timeouts when building java. The logs are producing 100s of GB of data due to an unnecessary debug log msg.

### What changes are included in this PR?

* Delete log message on write to memory

### Are these changes tested?

Yes, via conbench

### Are there any user-facing changes?

No
* GitHub Issue: #40775

### CODE REVIEW

**Comment by danepitkin** [comment]
@ursabot please benchmark lang=Java

File: dev/conbench_envs/benchmarks.env [hunk 1]
```diff
@@ -48,3 +48,4 @@ PARQUET_BUILD_EXAMPLES=ON
 PARQUET_BUILD_EXECUTABLES=ON
 PYTHON=python
 LD_LIBRARY_PATH=$CONDA_PREFIX/lib:$LD_LIBRARY_PATH
+JAVA_OPTIONS=-Xmx8G
```
Comment (vibhatha) [inline dev/conbench_envs/benchmarks.env:51]: @danepitkin just curious, should we set `Xms` as well?

Comment (vibhatha) [reply dev/conbench_envs/benchmarks.env:51]: Btw, would it be possible to make this CLI args? Just curious.

**Comment by danepitkin** [comment]
@ursabot please benchmark lang=Java

File: dev/conbench_envs/benchmarks.env (hunk 1 above)
Comment (austin3dickey) [reply dev/conbench_envs/benchmarks.env]: Thanks for trying this out, @danepitkin 🙂 If it turns out that tweaking the environment configuration fixes the error, I think leaving that configuration here (in the `dev/conbench_envs` location, which is only used by the Ursa machine) is the best choice. 

Reason is we support running the `voltrondata-labs/benchmarks` in other locations too, like local development, so I wouldn't want to always clobber the memory options of users' machines with a CLI arg there.

**Comment by danepitkin** [comment]
@ursabot please benchmark lang=Java

File: dev/conbench_envs/benchmarks.env (hunk 1 above)
Comment (danepitkin) [reply dev/conbench_envs/benchmarks.env]: Makes sense to me! Isolating the fix to the smallest scope is the preferred option IMO.

**Comment by danepitkin** [comment]
Just noting here that setting java max heap size to 8GB throws

```

Exception in thread "CommonsExecStreamPumper-pool-9-thread-2" java.lang.OutOfMemoryError: Java heap space
--
  | at java.lang.StringCoding.decode(StringCoding.java:215)
  | at java.lang.String.<init>(String.java:463)
  | at java.lang.String.<init>(String.java:515)
  | at com.gradle.d.a.a.a(SourceFile:83)
  | at com.gradle.d.a.a.flush(SourceFile:78)
  | at com.gradle.d.a.a.write(SourceFile:72)
  | at java.io.PrintStream.write(PrintStream.java:480)
  | at com.gradle.d.a.b.write(SourceFile:203)
  | at org.apache.commons.exec.StreamPumper.run(StreamPumper.java:112)
  | at java.lang.Thread.run(Thread.java:750)
```

but increasing max heap size to 16GB throws 

```

Exception in thread "CommonsExecStreamPumper-pool-9-thread-2" java.lang.OutOfMemoryError: GC overhead limit exceeded
--
  | at java.util.Arrays.copyOf(Arrays.java:3236)
  | at java.io.ByteArrayOutputStream.toByteArray(ByteArrayOutputStream.java:191)
  | at com.gradle.d.a.a.flush(SourceFile:78)
  | at com.gradle.d.a.a.write(SourceFile:72)
  | at java.io.PrintStream.write(PrintStream.java:480)
  | at com.gradle.d.a.b.write(SourceFile:203)
  | at org.apache.commons.exec.StreamPumper.run(StreamPumper.java:112)
  | at java.lang.Thread.run(Thread.java:750)
```

**Comment by danepitkin** [comment]
@ursabot please benchmark lang=Java

**Comment by danepitkin** [comment]
@ursabot please benchmark lang=Java

**Comment by danepitkin** [comment]
@ursabot please benchmark lang=Java

**Comment by danepitkin** [comment]
@ursabot please benchmark lang=Java

**Comment by danepitkin** [comment]
> Super exciting! Everything looks great to me.
> 
> Looks like the `alignBenchmark` got [almost 100x faster](https://conbench.ursa.dev/benchmark-results/0660df0498037e8380004e62ec039e68/), perhaps due to the logging change. All the other benchmarks are in line with what they were previously.

Amazing! I'll clean up this PR and then get it reviewed ASAP. Thanks @austin3dickey !

**Comment by danepitkin** [comment]
@ursabot please benchmark lang=Java

**Comment by danepitkin** [comment]
@ursabot please benchmark lang=Java

**Comment by danepitkin** [comment]
@ursabot please benchmark lang=Java

**Comment by danepitkin** [comment]
@kou @vibhatha this is ready for final review!

**Review by lidavidm — APPROVED** [review]

**Comment by lidavidm** [comment]
@danepitkin mind updating the PR description now that the approach is different?

**Comment by danepitkin** [comment]
> @danepitkin mind updating the PR description now that the approach is different?

Agh, good catch. Done!

**Review by kou — APPROVED** [review]
+1

**Pull request merged**